Vector.hpp

```diff
@@ -0,0 +1,246 @@
+#ifndef VECTOR_HPP
+
+#define VECTOR_HPP
+
+#include <iostream>
+
+
+template<typename T>
+class Vector
+{
+public:
+	/**
+	 * @brief Default constructor.
+	 *
+	 * Constructs an empty Vector with an initial capacity of 2.
+	 */
```

Comment:
no need to over-document and state the obvious. 
well written code mostly explains itself